pkg/operator/controller/controller.go

```diff
@@ -230,23 +223,24 @@ func shouldTakeUpdatePath(logger logr.Logger, targetVersion, currentVersion stri
 }
 
 func (r *ReconcileCDI) reconcileCreate(logger logr.Logger, cr *cdiv1alpha1.CDI) (reconcile.Result, error) {
-	MarkCrDeploying(cr, "DeployStarted", "Started Deployment")
 	// claim the configmap
 	if err := r.createConfigMap(cr); err != nil {
-		MarkCrFailed(cr, "ConfigError", "Unable to claim ConfigMap")
 		return reconcile.Result{}, err
 	}
 
 	logger.Info("ConfigMap created successfully")
 
+	MarkCrDeploying(cr, "DeployStarted", "Started Deployment")
+
 	if err := r.crInit(cr); err != nil {
-		MarkCrFailed(cr, "CrInitError", "Unable to Initialize CR")
 		return reconcile.Result{}, err
 	}
 
 	logger.Info("Successfully entered Deploying state")
 
-	return r.reconcileUpdate(logger, cr)
+	// since the CR was updated above, reconcile will be imediately triggered again
```

Comment:
Typo: immediately

Reply:
unit tests broke so had to go back to calling reconcileUpdate here